File: week_1/yoon/sw_4873.cpp
#include <iostream>
#include <stack>
#include <string>
using namespace std;

int solution(string line)
{
    stack<char> s;

    for(int i = 0; i < line.size(); ++i)
    {
        if(s.size() == 0) s.push(line[i]);
        else
        {
            if(s.top() == line[i]) s.pop();
            else s.push(line[i]);
        }
        
    }
    int length = s.size();

    return length;
}

int main()
{
    int t, result;
    cin >> t;
    string line;
    getline(cin, line);

    for(int i = 1; i <= t; ++i)
    {
        getline(cin ,line);
        result = solution(line);

        cout << "#" << i << " " << result << endl;
    }
    return 0;
}

File: PS_Study/week_01/yoon/Stack/sw_4873.py
T = int(input())
# 여러개의 테스트 케이스가 주어지므로, 각각을 처리합니다.
for test_case in range(1, T + 1):
    string = input()
    stack = []
    for s in string:
        if len(stack) == 0:
            stack.append(s)
        else:
            if stack[-1] == s:
                stack.pop()
            else:
                stack.append(s)
    result = len("".join(stack))
    
    print(f'#{test_case} {result}')
                